### PR TITLE
#4133 - SFAS to SIMS Bridge - PT App Cancellation Date & MSFAA (FT/PT) Change - Part 3

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- Changed logic to: 
  - Not consider part-time SFAS cancelled applications for program year totals; 
  - Not consider part-time application date overlap and; 
  - Ignore full-time and part-time applications in the MSFAA number process.
- Added E2E tests;

### CODE REVIEW

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts [hunk 1]
```diff
@@ -825,6 +825,112 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
...
+          endDate: getISODateOnlyString(legacyApplicationEndDate),
+          csgdAward: 9,
+          csgpAward: 10,
+          sbsdAward: 12,
+          bcagAward: 13,
+          applicationCancelDate: getISODateOnlyString(new Date()),
+        },
+      },
+    );
+    await db.sfasApplication.save([fakeSFASApplication]);
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts:899]: Minor. `await db.sfasApplication.save(fakeSFASApplication);` can be done

File: sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts [hunk 2]
```diff
@@ -825,6 +825,112 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
...
+          csptAward: 2,
+          csgdAward: 3,
+          csgpAward: 4,
+          sbsdAward: 6,
+          bcagAward: 7,
+          applicationCancelDate: getISODateOnlyString(new Date()),
+        },
+      },
+    );
+    await db.sfasPartTimeApplications.save([fakeSFASPartTimeApplication]);
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts:916]: same here. https://github.com/bcgov/SIMS/pull/4182/files#r1898767268

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts [hunk 3]
```diff
@@ -449,6 +449,71 @@ describe("DisbursementController(e2e)-associateMSFAA", () => {
...
+    );
+
+    // Fetch MSFAA Number for the student in SIMS.
+    const [createdMSFAANumber] = await db.msfaaNumber.find({
+      select: {
+        msfaaNumber: true,
+      },
+      relations: {
+        referenceApplication: true,
+      },
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts:504]: Do we require the relations here?

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts [hunk 4]
```diff
@@ -449,6 +449,71 @@ describe("DisbursementController(e2e)-associateMSFAA", () => {
...
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the student in SIMS.
+    const [createdMSFAANumber] = await db.msfaaNumber.find({
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts:498]: findOne?

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts [hunk 5]
```diff
@@ -449,6 +449,71 @@ describe("DisbursementController(e2e)-associateMSFAA", () => {
...
+    const legacyApplicationEndDate = addDays(-10);
+
+    // Create student and save fake application with disbursements.
+    const student = await saveFakeStudent(db.dataSource);
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+        applicationStatus: ApplicationStatus.Completed,
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts:464]: Minor. I would advice to keep the application status to `In-Progress` to reflect close to real time data.

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts [hunk 6]
```diff
@@ -617,6 +682,71 @@ describe("DisbursementController(e2e)-associateMSFAA", () => {
...
+    const legacyApplicationEndDate = addDays(-10);
+
+    // Create student and save fake application with disbursements.
+    const student = await saveFakeStudent(db.dataSource);
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts:697]: Minor. I would advice to keep the application status to `In-Progress` to reflect close to real time data.

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts [hunk 7]
```diff
@@ -617,6 +682,71 @@ describe("DisbursementController(e2e)-associateMSFAA", () => {
...
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the student in SIMS.
+    const [createdMSFAANumber] = await db.msfaaNumber.find({
+      select: {
+        msfaaNumber: true,
+      },
+      relations: {
+        referenceApplication: true,
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts:736]: Same.

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts [hunk 8]
```diff
@@ -617,6 +682,71 @@ describe("DisbursementController(e2e)-associateMSFAA", () => {
...
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the student in SIMS.
+    const [createdMSFAANumber] = await db.msfaaNumber.find({
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts:731]: findone

**Comment by dheepak-aot** [comment]
Great Work. Only minor comments.

File: sources/packages/backend/libs/services/src/sfas/sfas-part-time-application.service.ts [hunk 9]
```diff
@@ -38,7 +38,10 @@ export class SFASPartTimeApplicationsService extends DataModelService<SFASPartTi
       .createQueryBuilder("sfasPTApplication")
       .select(["sfasPTApplication.id"])
       .innerJoin("sfasPTApplication.individual", "sfasPTstudent")
-      .where("lower(sfasPTstudent.lastName) = lower(:lastName)", { lastName })
+      .where("sfasPTApplication.applicationCancelDate IS NULL")
```
Comment (guru-aot) [inline sources/packages/backend/libs/services/src/sfas/sfas-part-time-application.service.ts:41]: 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrepestana-aot

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes. Looks good 👍

**Pull request merged**